services/traces/engine.go

```diff
@@ -126,6 +121,18 @@ func (e *Engine) indexBlockTraces(evmBlock *types.Block, cadenceBlockID flow.Ide
 	wg.Wait()
 }
 
+// ID is required by the publisher interface and we return a random uuid since the
+// subscription will only happen once by this engine
+func (e *Engine) ID() uuid.UUID {
+	return uuid.New()
```

Comment:
I don't think we need to, because this subscription will happen only once, and if it's closed a new one will be generated upon reconnecting. Tbh this method is "dummy" as we just need it to comply with interface

Reply:
Sounds good then.